rust/chains/hyperlane-ethereum/src/rpc_clients/fallback.rs

```diff
@@ -98,9 +171,83 @@ impl From<FallbackError> for ProviderError {
     }
 }
 
+async fn handle_stalled_provider<C>(
+    fallback_provider: &FallbackProvider<C>,
+    priority: &PrioritizedProviderInner,
+) -> Result<(), ProviderError>
+where
+    C: JsonRpcClient,
+{
+    let now = Instant::now();
+    if now
+        .duration_since(priority.last_block_height.1)
+        .le(&fallback_provider.max_block_time)
+    {
+        // Do nothing, it's too early to tell if the provider has stalled
+        return Ok(());
+    }
+
+    let provider = &fallback_provider.inner.providers[priority.index];
```

Comment:
I don't feel strongly but noting that I think we could just make this a param to `handle_stalled_provider` as we did this lookup in the call site already